docs/compilation/performance_tips.md

```diff
@@ -0,0 +1,505 @@
+# Performance tips
+
+The most costly operation in Concrete is the table lookup operation, so the primary goal of optimizing performance is to reduce the amount of table lookups and reduce the bit width of the inputs to table lookups.
```

Comment:
Maybe it could be good here to have some numbers (or plots), saying that the inputs bitwidth of table lookup is the *main* factor which affect the complexity of table lookup, so we can illustrate:

```
from concrete import fhe
import matplotlib.pyplot as plot
import numpy as np


def get_complexity_per_bitwitdh(bitwidth):
    def f(x):
        return fhe.univariate(lambda x : x ^ 2)(x)
    circuit = fhe.Compiler(f, {"x": "encrypted"}).compile(inputset)
    print(range(2**bitwidth), circuit.complexity)
    print(circuit.programmable_bootstrap_count_per_parameter)
    return circuit.complexity


complexity_per_bitwitdh = map(get_complexity_per_bitwitdh, range(16))

plot.plot(np.array(list(complexity_per_bitwitdh)))
plot.show()
```

and insert the graph here, that can give a good intuition. So the first step to optimize an fhe programs is to try to reduce the bitwidth a maximum maybe a link to some ml quantization as example?

Reply:
I've added it for 2-8 bit range as bigger bits would cause confusion due to encoding as discussed in slack. As for ML quantization examples, wdyt @andrei-stoian-zama, any documentation page I can use to highlight the importance of bit width of tlu inputs?

Reply:
I can't think of a documentation page to link, I don't think the relationship between bitwidth and complexity is explained somewhere. I think in general we say n_bits ~ complexity.

Reply:
There are more variables to complexity, but it's okay, we already said in the documentation that performance is correlated to table lookup input bit width, it's all good IMO